lib/backend/firestore/firestorebk.go

```diff
@@ -238,17 +235,28 @@ func CreateFirestoreClients(ctx context.Context, projectID string, endPoint stri
 	return firestoreAdminClient, firestoreClient, nil
 }
 
+// Option supplies a non-config related option to a Backend during
+// construction
+type Option func(*Backend)
+
+// WithClock provides a custom clock the firestore Backend
+func WithClock(clock clockwork.Clock) Option {
+	return func(b *Backend) {
+		b.clock = clock
+	}
+}
+
```

Comment:
If you are making this change, just follow passing a config struct approach we're doing now. You might want to look at how DynamoDB events backend is now.

Reply:
Done. `Config` name was already taken, so it's named `Options` instead.